index.bs

```diff
@@ -365,6 +368,8 @@ OBUs stored in the configOBUs field follow the [=open_bitstream_unit=] [=Low Ove
 
 <assert>When a [=Sequence Header OBU=] is contained within the configOBUs of the AV1CodecConfigurationRecord, the values present in the [=Sequence Header OBU=] contained within configOBUs SHALL match the values of the AV1CodecConfigurationRecord.</assert>
 
+<assert>When the samples associated with a sample entry do not contain any sync sample, a [=Sequence Header OBU=] SHALL be present in the [=configOBUs=] field of the AV1CodecConfigurationRecord of that sample entry.</assert>
```

Comment:
Maybe we can remove that change and the PR could be merged more easily

Reply:
Should we add a sentence saying that SH OBUs are typically not present in non-sync samples, even if it is not forbidden? It seems too late to forbid.

Reply:
Proposed new text:

```suggestion
<assert>When none of the samples associated with a sample entry contains a Sequence Header OBU, a [=Sequence Header OBU=] SHALL be present in the [=configOBUs=] field of the AV1CodecConfigurationRecord of that sample entry.</assert>
NOTE: For example, consider a AV1 stream encoded with a unique Key Frame at the star, followed by non Key Frames, and delivered over MPEG-2 Transport Stream (or equivalent). If recording into an MP4 does not happen from the start of the stream, none of the samples will contain a Sequence Header OBU, and therefore the resulting MP4 file cannot be decoded at all. If the Sequence Header OBU is obtained by other means than looking at the samples (e.g. using MPEG-2 tables), even if the recorded stream is not strictly speaking valid according to AV1 because it does not start with a Key Frame, some robust decoders may be able to process it.
```

Reply:
We should study how RTP delivers SH OBUs.

Reply:
Instead of "when none of samples" should it be "when the first sample does not contain ..."?

Reply:
> We should study how RTP delivers SH OBUs.

A quick review of [the RTP spec](https://aomediacodec.github.io/av1-rtp-spec/) seems to indicate that SH OBUs are not delivered out of band, at least there is no mechanism defined. The spec says:
> If a sequence header OBU is present in an RTP packet, then it SHOULD be the first OBU in the packet. 

So the use case envisaged in this AV1-ISOBMFF seems only really relevant for MPEG-2 TS cases.